docs/release-process.md

```diff
@@ -33,18 +33,25 @@ To create a new release, follow these steps:
 5. Create a release branch with a name that follows the `release-X.Y` format.
```

Comment:
sidenote: should this doc live under `docs/developer`?

Reply:
> sidenote: should this doc live under docs/developer?

This makes sense. However, the Versioning and Release Planning and Development sections are for the user also,  so perhaps we can split into two docs?

Reply:
added an issue https://github.com/nginxinc/nginx-kubernetes-gateway/issues/896